### ISSUE TITLE
Missed quoting recommendation

### ISSUE THREAD

**Issue by PhilZ-cwm6** [koalaman/shellcheck#2401]
#### For bugs
- Rule Id (if any, e.g. SC1000): 
- My shellcheck version (`shellcheck --version` or "online"): 
- [ ] The rule's wiki page does not already cover this (e.g. https://shellcheck.net/wiki/SC2086)
- [ ] I tried on https://www.shellcheck.net/ and verified that this is still a problem on the latest commit

#### For new checks and feature suggestions
- [ X] https://www.shellcheck.net/ (i.e. the latest commit) currently gives no useful warnings about this
- [ X] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```
#!/bin/sh
getStr() {
    echo "$1"
}
var="*"
mypath=$(getStr ${1:+$var})

echo "$mypath"
```

script.sh hello
#### Here's what shellcheck currently says:
It lists the first file in current directory: globbing

#### Here's what I wanted or expected to see:
War about globbing if $var is unquoted

**Comment by Flupp** [comment]
I noticed a similar problem within command substitution within `[[ … ]]` when no quoting for `$(…)` is needed, however, within it, quoting is still necessary.

Consider the following example:
```bash
x='a  b'
[[ $(echo "$x") == 'a  b' ]]; echo $?
```
Note the two spaces between `a` and `b`. This example outputs 0 as expected. However, the following outputs 1 (as expected but probably usually not intended).
```bash
x='a  b'
[[ $(echo $x) == 'a  b' ]]; echo $?
```
I’d expect a warning about the missing quotes around `$x`.
